packages/@aws-cdk/aws-eks/lib/managed-nodegroup.ts

```diff
@@ -291,17 +311,21 @@ export class Nodegroup extends Resource implements INodegroup {
       tags: props.tags,
     });
 
+    if (props.capacityType) {
+      resource.addPropertyOverride('CapacityType', props.capacityType.valueOf());
+    }
+
     if (props.launchTemplateSpec) {
       if (props.diskSize) {
         // see - https://docs.aws.amazon.com/eks/latest/userguide/launch-templates.html
         // and https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-eks-nodegroup.html#cfn-eks-nodegroup-disksize
         throw new Error('diskSize must be specified within the launch template');
       }
-      if (props.instanceType) {
```

Comment:
Why did you remove this validation?

Reply:
According to this:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-eks-nodegroup.html#cfn-eks-nodegroup-instancetypes

>If you specify launchTemplate, then you can specify zero or one instance type in your launch template or you can specify 0-20 instance types for instanceTypes. If however, you specify an instance type in your launch template and specify any instanceTypes, the node group deployment will fail. If you don't specify an instance type in a launch template or for instanceTypes, then t3.medium is used, by default. If you specify Spot for capacityType, then we recommend specifying multiple values for instanceTypes.

If we design the `Nodegroup` API like

```ts
new Nodegroup(..., {
   instanceType: ec2.InstanceType,
   additionalInstanceTypes: ec2.InstanceType[],
   launchTemplateSpec,
});
```

In this case, one or multipe instance types may also be defined in  the`launchTemplate` and this will cause failure according to the document. However, This should also be accepted as user may leverage some other  features from LT while still defining the multiple instance types in `additionalInstanceTypes`.

As we can't check the content of the LT passed in and we encourage users to define instance types in `additionalInstanceTypes`, we probably should skip the check. Ideas?

Reply:
Ok so basically a user can specify a launch template without any instance types, and only specify them in the `InstanceTypes` property of the nodegroup, in which the deployment will succeed right? If thats the case then yes we should remove this validation 👍